apps/emqx_rule_engine/src/emqx_rule_engine.erl

```diff
@@ -507,7 +507,7 @@ rule_id() ->
     gen_id("rule:", fun emqx_rule_registry:get_rule/1).
 
 gen_id(Prefix, TestFun) ->
-    Id = iolist_to_binary([Prefix, emqx_plugin_libs_id:gen()]),
+    Id = iolist_to_binary([Prefix, emqx_misc:gen_id()]),
     case TestFun(Id) of
         not_found -> Id;
         _Res -> gen_id(Prefix, TestFun)
```

Comment:
why deadloop here ?

Reply:
ah, never mind, forgot about the random part.

Reply:
If the TestFun/1 returns something, then the id already exists. In this case we try again to create a new random id.